HARK/core.py

```diff
@@ -594,10 +606,46 @@ def make_shock_history(self):
             (self.T_sim, self.AgentCount), dtype=bool
         )
 
+        # Also make blank arrays for the draws of newborns' initial conditions
+        for var_name in self.state_vars:
+            self.newborn_init_history[var_name] = (
+                np.zeros((self.T_sim, self.AgentCount)) + np.nan
+            )
+
+        # Record the initial condition of the newborns created by
+        # initialize_sim -> sim_births
+        for var_name in self.state_vars:
+            # Check whether the state is array-like or an aggregate
+            if hasattr(self.state_now[var_name], "shape"):
+                self.newborn_init_history[var_name][self.t_sim] = self.state_now[
+                    var_name
+                ]
+            else:
+                self.newborn_init_history[var_name][self.t_sim] = self.state_now[
+                    var_name
+                ]
```

Comment:
Looks like the same thing is being done in either condition here.

Reply:
Good catch, thanks